### PR TITLE
api: timeSeries proto

### PR DESCRIPTION
**Pull request by smonero**
### Description

We need a timeseries proto to represent events that will be put on the timeline.

One thing I thought about and I don't have a good answer for is how we would prioritize / visualize overlapping events. I guess we will figure that out in the future.

### Testing Performed

none

### GitHub Issue

Fixes #

### CODE REVIEW

File: api/timeline/v1/timeseries.proto [hunk 1]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package clutch.timeline.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/timeline/v1;timelinev1";
+
+import "google/protobuf/any.proto";
+import "validate/validate.proto";
+
+message TimeseriesData {
```
Comment (mikecutalo) [inline api/timeline/v1/timeseries.proto:10]: Can you write a comment about what this represents?  It looks like just a single datapoint but some more color here would be helpful to the reader.

File: api/timeline/v1/timeseries.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  string project_name = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  // For a range, both start time and end time will be present
+  // For a singular event, only start time will be filled in.
+  int64 start_time_millis = 2 [ (validate.rules).int64.gt = 0 ];
+  int64 end_time_millis = 3;
```
Comment (mikecutalo) [inline api/timeline/v1/timeseries.proto:15]: What if the event is just a moment in time and not a range, how would handle that situation?

File: api/timeline/v1/timeseries.proto [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  int64 event_id = 4;
```
Comment (mikecutalo) [inline api/timeline/v1/timeseries.proto:16]: What will this be used for?

File: api/timeline/v1/timeseries.proto (hunk 2 above)
Comment (smonero) [reply api/timeline/v1/timeseries.proto:15]: My plan for that was end_time_millis would be 0

File: api/timeline/v1/timeseries.proto [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+option go_package = "github.com/lyft/clutch/backend/api/timeline/v1;timelinev1";
+
+import "google/protobuf/any.proto";
+import "validate/validate.proto";
+
+// A timeseries datapoint message is useful for organizing events
+// to be displayed in a timeline view.  Users can transform data
+// into a timeseries format and be able to organize them.
+message TimeseriesData {
+  string project_name = 1 [ (validate.rules).string = {min_bytes : 1} ];
```
Comment (dschaller) [inline api/timeline/v1/timeseries.proto:14]: should we call this project name? It seems like time series data could be used for things even outside of projects.

File: api/timeline/v1/timeseries.proto (hunk 2 above)
Comment (dschaller) [reply api/timeline/v1/timeseries.proto:18]: They could also be equal. e.g. a moment is just the start and end at the same time

File: api/timeline/v1/timeseries.proto [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  // For a range, both start time and end time will be present
+  // For a singular event, only start time will be filled in.
+  int64 start_time_millis = 2 [ (validate.rules).int64.gt = 0 ];
+  int64 end_time_millis = 3;
+  // The plan here is to have each user of this proto to put their unique data in a proto
```
Comment (dschaller) [inline api/timeline/v1/timeseries.proto:19]: nit: probably would change this to either be more concrete or to be written as a TODO

File: api/timeline/v1/timeseries.proto [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+// A timeseries Point message is useful for organizing events
+// to be displayed in a timeline view.  Users can transform data
+// into a timeseries format and be able to organize them.
+message Point {
+  oneof event {
+    option (validate.required) = true;
+    TimeRange range = 2;
+    int64 singular_event_timestamp_millis = 3;
+  }
```
Comment (danielhochman) [inline api/timeline/v1/timeseries.proto:23]: nit: naming
```suggestion
  oneof timestamp {
    option (validate.required) = true;
    TimeRange range = 2;
    int64 millis = 3;
  }
```

File: api/timeline/v1/timeseries.proto [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  google.protobuf.Any event_info = 5;
+  string event_description = 6;
```
Comment (danielhochman) [inline api/timeline/v1/timeseries.proto:25]:
```suggestion
  google.protobuf.Any pb = 4;
  string description = 5;
```

(to more closely match https://github.com/lyft/clutch/blob/498410f57fc1b534d941ec2473e1abf3e58d5609/api/topology/v1/topology_api.proto#L224)

File: api/timeline/v1/timeseries.proto [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+package clutch.timeseries.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/timeseries/v1;timeseriesv1";
+
+import "google/protobuf/any.proto";
+import "validate/validate.proto";
+
+message TimeRange {
+  int64 start_time_millis = 1 [ (validate.rules).int64.gt = 0 ];
+  int64 end_time_millis = 2 [ (validate.rules).int64.gt = 0 ];
```
Comment (danielhochman) [inline api/timeline/v1/timeseries.proto:12]: nit verbosity
```suggestion
  int64 start_millis = 1 [ (validate.rules).int64.gt = 0 ];
  int64 end_millis = 2 [ (validate.rules).int64.gt = 0 ];
```

**Review by danielhochman — COMMENTED** [review]
some naming nits, saving chars will actually pay big dividends when requesting huge timeseries sets in JSON format

**Review by mikecutalo — APPROVED** [review]
Looks good I dont have any additional comments atm.

File: api/timeline/v1/timeseries.proto [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+// A timeseries Point message is useful for organizing events
+// to be displayed in a timeline view.  Users can transform data
+// into a timeseries format and be able to organize them.
+message Point {
+  oneof timestamp {
+    option (validate.required) = true;
+    TimeRange range = 2;
+    int64 millis = 3;
+  }
+  google.protobuf.Any pb = 5;
```
Comment (danielhochman) [inline api/timeline/v1/timeseries.proto:24]: `4` is missing

Comment (danielhochman) [reply api/timeline/v1/timeseries.proto:24]: actually so is `1`, renumber plz

File: api/timeline/v1/timeseries.proto [hunk 10]
```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
```
Comment (danielhochman) [inline api/timeline/v1/timeseries.proto:1]: folder name is still timeline, should be timeseries

**Pull request merged**